com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/VariablesRequestHandler.java

```diff
@@ -305,14 +323,21 @@
             } else if (sizeValue != null) {
                 detailsValue = "size=" + variableFormatter.valueToString(sizeValue, options);
             } else if (DebugSettings.getCurrent().showToString) {
-                try {
-                    detailsValue = VariableDetailUtils.formatDetailsValue(value, containerNode.getThread(), variableFormatter, options, evaluationEngine);
-                } catch (OutOfMemoryError e) {
-                    logger.log(Level.SEVERE, "Failed to compute the toString() value of a large object", e);
-                    detailsValue = "<Unable to display the details of a large object>";
-                } catch (Exception e) {
-                    logger.log(Level.SEVERE, "Failed to compute the toString() value", e);
-                    detailsValue = "<Failed to resolve the variable details due to \"" + e.getMessage() + "\">";
+                if (VariableDetailUtils.isLazyLoadingSupported(value) && varProxy != null) {
+                    typedVariables.presentationHint = new VariablePresentationHint(true);
+                    VariableProxy valueReferenceProxy = this.getVariableProxy(containerNode, value, "", indexedVariables, javaVariable);
+                    Integer referencedVariableId = context.getRecyclableIdPool().addObject(containerNode.getThreadId(), valueReferenceProxy);
+                    varProxy.setReferencedVariableId(referencedVariableId);
```

Comment:
No need to regenerate a VariableProxy, just use a new property `isLazyResolved` to mark this variable to be lazy resolved.